tests/utils/em.py

```diff
@@ -0,0 +1,313 @@
+    """
+    Get the list of powers for a given node.
+    :param node: A node of the energy model. It can be a cluster or cpu.
+    :type node: EnergyModelNode
+
+    :returns: The list of powers associates with the given node.
+    """
+    powers = []
+    for freq in node.active_states.keys():
+        powers.append(node.active_states[freq].power)
+    return powers
+
+def _get_capacities_list(node):
+    """
+    Get the list of capacities for a given node.
+    :param node: A node of the energy model. It can be a cluster or cpu.
+    :type node: EnergyModelNode
+
+    :returns: The list of capacities associates with the given node.
+    """
+    cap = []
+    for freq in node.active_states.keys():
+        cap.append(node.active_states[freq].capacity)
+    return cap
+
+def _is_list_strickly_increasing(l):
+    return all(x<y for x, y in zip(l, l[1:]))
+
+def is_power_increasing(energy_model):
+    """
+    Check that for all the nodes of the energy model the power is an increasing
+    sequence.
+    :params energy_model: The energy model to control
+    :type energy_model: EnergyModel
+
+    :returns: A tuple that contains True if all nodes respect the
+              previous assumptions, False otherwise; and an error message.
+    """
+    powers = []
+    msg = ''
+    failed_nodes = []
+    succeed = True
+    for cluster in energy_model.root.children:
+        power = _get_powers_list(cluster)
+        # Check that clusters powers are strickly increasing
+        if not _is_list_strickly_increasing(power):
+            failed_nodes.append(cluster.name)
+            succeed = False
+
+    for cpu in energy_model.cpu_nodes:
+        powers = _get_powers_list(cpu)
+        # Check that cpu powers are strickly increasing
+        if not _is_list_strickly_increasing(powers):
+            failed_nodes.append(cpu.name)
+            succeed = False
+
+    if not succeed:
+        msg = 'The power lists is not increasing for {}'.format(failed_nodes)
+    return (succeed, msg)
+
+def is_efficiency_decreasing(energy_model):
+    """
+    Check that for all the cpus of the energy model the energy efficiency is
+    decreasing.
+    :params energy_model: The energy model to control
+    :type energy_model: EnergyModel
+
+    :returns: A tuple that contains True if the energy_model respects
+              the previous assumptions, False otherwise; and an error message.
+    """
+    msg = ''
+    powers = []
+    cap = []
+    energy = []
+    failed_nodes = []
+    succeed = True
+    for cpu in energy_model.cpu_nodes:
+        powers = _get_powers_list(cpu)
+        cap = _get_capacities_list(cpu)
+        energy = [float(powers[i])/cap[i] for i in range(len(powers))]
+        # Check that power / cap is strickly increasing which is equivalent
+        # to check that the cap / power (energy efficiency) is decreasing
+        if not _is_list_strickly_increasing(energy):
+            failed_nodes.append(cpu.name)
+            succeed = False
```

Comment:
The goal here was to get the list of cpus for which the energy model does not respect the property. If i add the break there it would only return the first faulty cpu.